src/fixit/rules/deprecated_abc_import.py

```diff
@@ -0,0 +1,331 @@
+            """,
+        ),
+    ]
+
+    def __init__(self) -> None:
+        super().__init__()
+        # If the module needs to updated
+        self.update_module: bool = False
+        # The original imports
+        self.imports_names: List[str] = []
+
+    def visit_ImportFrom(self, node: cst.ImportFrom) -> None:
+        """
+        This catches the `from collections import <ABC>` cases
+        """
+        # Get imports in this statement
+        import_names = (
+            [name.name.value for name in node.names]
+            if type(node.names) is tuple
+            else []
+        )
+        # Filter the imports for ABC imports
+        import_names_in_abc = [name in ABCS for name in import_names]
+        if (
+            node.module
+            and node.module.value == "collections"
+            and any(import_names_in_abc)
+        ):
+            # Replacing the case where there are ABCs mixed with non-ABCs requires
+            # splitting a single import statement into two separate imports. This
+            # cannot be achieved in this method and is offloaded to leaving the module.
+            if not all(import_names_in_abc):
+                # We set this variable which triggers the `self.report` to be called
+                # in `leave_Module`. We report in the `leave_Module`
+                # so that we can add an additional `SimpleStatementLine` for the new
+                # import
+                self.update_module = True
+                self.imports_names = import_names
+            else:
+                self.report(
+                    node,
+                    replacement=node.with_changes(
+                        module=cst.Attribute(
+                            value=cst.Name(value="collections"),
+                            attr=cst.Name(value="abc"),
+                        )
+                    ),
+                )
+
+    def get_import_from(
+        self, node: Union[cst.SimpleStatementLine, cst.BaseCompoundStatement]
+    ) -> Optional[cst.ImportFrom]:
+        """
+        Iterate over a Statement Sequence and return a Statement if it is a
+        `cst.ImportFrom` statement.
+        """
+        if m.matches(
+            node,
+            m.SimpleStatementLine(
+                body=[
+                    m.ZeroOrMore(),
+                    m.ImportFrom(
+                        module=m.Name("collections"),
+                        names=m.OneOf(
+                            [m.ImportAlias(name=m.Name(n)) for n in self.imports_names]
+                        ),
+                    ),
+                    m.ZeroOrMore(),
+                ]
+            ),
+        ):
+            imp = m.findall(
+                node,
+                m.ImportFrom(
+                    module=m.Name("collections"),
+                    names=m.OneOf(
+                        [m.ImportAlias(name=m.Name(n)) for n in self.imports_names]
+                    ),
+                ),
+            )[0]
```

Comment:
Do you need the conditional `m.matches()` guarding this? Maybe just `m.findall()` and have a conditional on whether it finds something?